components/brave_wallet_ui/common/slices/api.slice.ts

```
@@ -2676,6 +2693,140 @@ export function createWalletApi () {
+                    ? 'SPL'
+                    : '',
+                tokenID: arg.tokenId,
+                imageURL: response.image.startsWith('data:image/')
+                  ? response.image
+                  : await translateToNftGateway(response.image),
+                imageMimeType: 'image/*',
+                floorFiatPrice: '',
+                floorCryptoPrice: '',
+                contractInformation: {
+                  address: arg.contractAddress,
+                  name: response.name,
+                  description: response.description,
+                  website: '',
+                  facebook: '',
+                  logo: '',
+                  twitter: ''
+                },
+                attributes
+              }
+
+              return {
+                data: nftMetadata
+              }
+            } else {
+              console.error('Error fetching NFT metadata: ', result.error, result.errorMessage)
+              throw new Error(result.errorMessage)
+            }
+          }
+          catch (error) {
+            console.error('Error fetching NFT metadata: ', error)
+            throw error
+          }
+        },
+        providesTags: (_result, err, arg) => err
+        ? ['ERC721Metadata']
+        : [{ type: 'ERC721Metadata', id: getAssetIdKey(arg) }] 
+      }),
+      getNftPinningStatus: query<BraveWallet.TokenPinStatus | undefined, BraveWallet.BlockchainToken>({
+        queryFn: async (arg, _api, _extraOptions, baseQuery) => {
+          try {
+            const { data: api } = baseQuery(undefined)
+            const { braveWalletPinService} = api
+            const result = await braveWalletPinService.getTokenStatus(arg)
+
+            if (result.error) {
+              return {
+                error: result.error.message
+              }
+            } else {
+              return {
+                data: result.status ? result.status.local : undefined
```

might need to throw an error here